lib/ace/edit_session/bracket_match.js

```diff
@@ -117,7 +117,7 @@ function BracketMatch() {
             typeRe = new RegExp(
                 "(\\.?" +
                 token.type.replace(".", "\\.").replace("rparen", ".paren")
-                    .replace(/\b(?:end|start|begin)\b/, "")
```

Comment:
Modes imported from textmate have token types like `punctuation.begin.languagename`  `punctuation.start.langaugename` `punctuation.end.langaugename`. 
This is meant to create a regexp `punctuation\.(?:start|begin)\.languagename` but i see there's still a  bug, correct version would be `punctuation\.(?:start|begin|end)\.languagename`